src/main/java/org/springframework/data/convert/PropertyValueConverterRegistrar.java

```diff
@@ -0,0 +1,191 @@
+	 * Starts a converter registration by pointing to a property of a domain type.
+	 *
+	 * @param <T> the domain type
+	 * @param <S> the property type
+	 * @param type the domain type to obtain the property from
+	 * @param property a function to describe the property to be referenced. Usually a method handle to a getter.
+	 * @return will never be {@literal null}.
+	 */
+	public <T, S> WritingConverterRegistrationBuilder<T, S, P> registerConverter(Class<T> type, Function<T, S> property) {
+
+		String propertyName = MethodInvocationRecorder.forProxyOf(type).record(property).getPropertyPath()
+				.orElseThrow(() -> new IllegalArgumentException("Cannot obtain property name!"));
+
+		return new WritingConverterRegistrationBuilder<T, S, P>(type, propertyName, this);
+	}
+
+	/**
+	 * Starts a converter registration by pointing to a property of a domain type.
+	 *
+	 * @param <T> the domain type
+	 * @param <S> the property type
+	 * @param type the domain type to obtain the property from
+	 * @param propertyName a function to describe the property to be referenced. Usually a method handle to a getter.
+	 * @return will never be {@literal null}.
+	 */
+	public <T, S> WritingConverterRegistrationBuilder<T, S, P> registerConverter(Class<T> type, String propertyName,
+			Class<S> propertyType) {
+		return new WritingConverterRegistrationBuilder<T, S, P>(type, propertyName, this);
+	}
+
+	/**
+	 * Register the given converter for the types property identified via its name.
+	 *
+	 * @param type the domain type to obtain the property from
+	 * @param path the property name.
+	 * @param converter the converter to apply.
+	 * @return this.
+	 */
+	public PropertyValueConverterRegistrar registerConverter(Class<?> type, String path,
+			PropertyValueConverter<?, ?, ? extends ValueConversionContext<?>> converter) {
+
+		registry.registerConverter(type, path,
+				(PropertyValueConverter<?, ?, ? extends ValueConversionContext<P>>) converter);
+		return this;
+	}
+
+	/**
+	 * Obtain the {@link SimplePropertyValueConverterRegistry}.
+	 *
+	 * @return new instance of {@link SimplePropertyValueConverterRegistry}.
+	 */
+	public ValueConverterRegistry<P> buildRegistry() {
+		return new SimplePropertyValueConverterRegistry<>(registry);
+	}
+
+	/**
+	 * Register collected {@link PropertyValueConverter converters} within the given {@link ValueConverterRegistry
+	 * registry}.
+	 *
+	 * @return new instance of {@link SimplePropertyValueConverterRegistry}.
+	 */
+	public void registerConvertersIn(ValueConverterRegistry<P> target) {
+
+		Assert.notNull(target, "Target registry must not be null!");
+
+		registry.getConverterRegistrationMap().entrySet().forEach(entry -> {
+			target.registerConverter(entry.getKey().type, entry.getKey().path, entry.getValue());
+		});
+	}
+
+	/**
+	 * Helper to build up a fluent registration API starting on
+	 *
+	 * @author Oliver Drotbohm
+	 */
+	static class WritingConverterRegistrationBuilder<T, S, P extends PersistentProperty<P>> {
```

Comment:
We need to increase visibility so that builders can be used from outside of the package.